engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -199,11 +222,23 @@ class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
     context.actorOf(props, s"ejha_for_$jobDescriptor")
   }
 
-  def lookupCachedResult(jobDescriptor: BackendJobDescriptor, cacheResultId: MetaInfoId) = {
-    // TODO: Start up a backend job copying actor (if possible, otherwise just runJob). That should send back the BackendJobExecutionResponse
-    self ! FailedNonRetryableResponse(jobKey, new Exception("Call cache result copying not implemented!"), None)
-    // While the cache result is looked up, we wait for the response just like we were waiting for a Job to complete:
-    goto(RunningJob) using EmptyPartialCompletionData
+  def lookupCacheResult(jobDescriptor: BackendJobDescriptor, bjeaProps: Props, taskOutputs: Seq[TaskOutput], cacheHit: CacheHit) = {
+    context.actorOf(FetchCachedResultsActor.props(cacheHit, taskOutputs, self, new CallCache(CromwellDatabase.databaseInterface)))
+    goto(FetchingCachedOutputsFromDatabase) using JobDescriptorData(jobDescriptor, bjeaProps)
+  }
+
+  def copyCacheHit(cacheHit: CacheHit, jobDescriptor: BackendJobDescriptor, bjeaProps: Props, cachedJobOutputs: JobOutputs) = {
+    factory.cacheHitCopyingActorProps(jobDescriptor, initializationData, serviceRegistryActor, cachedJobOutputs) match {
+      case Some(props) =>
+        context.actorOf(props, buildCacheHitCopyingActorName(jobDescriptor))
+        goto(CopyingCachedOutputs) using PartialCompletionDataWithJobResults(jobDescriptor, bjeaProps, cachedJobOutputs)
+      case None =>
```

Comment:
Is it worth short-circuiting the `fetchFromDatabase` stage too if this is `None`? Currently we fetch everything from the DB regardless and only once we get here decide to throw it all away